src/OrchardCore/OrchardCore.Media.Core/Services/MediaSizeLimitAttribute.cs

```diff
@@ -55,7 +55,7 @@ public void OnAuthorization(AuthorizationFilterContext context)
                 }
 
                 var effectiveRequestSizePolicy = context.FindEffectivePolicy<IRequestSizePolicy>();
-                if (effectiveRequestSizePolicy == null || effectiveRequestSizePolicy == this)
```

Comment:
This equality cannot be true currently as `this` is not `IRequestSizePolicy`, seemed weird and caused analysis warning (as error)